homeassistant/components/emoncms/config_flow.py

```diff
@@ -0,0 +1,170 @@
+    return [
+        {
+            "value": feed[FEED_ID],
+            "label": f"{feed[FEED_ID]}|{feed[FEED_TAG]}|{feed[FEED_NAME]}",
+        }
+        for feed in feeds
+    ]
+
+
+def sensor_name(url: str) -> str:
+    """Return sensor name."""
+    sensorip = url.rsplit("//", maxsplit=1)[-1]
+    return f"emoncms@{sensorip}"
+
+
+async def get_feed_list(hass: HomeAssistant, url: str, api_key: str) -> dict[str, Any]:
+    """Check connection to emoncms and return feed list if successful."""
+    emoncms_client = EmoncmsClient(
+        url,
+        api_key,
+        session=async_get_clientsession(hass),
+    )
+    return await emoncms_client.async_request("/feed/list.json")
+
+
+class EmoncmsConfigFlow(ConfigFlow, domain=DOMAIN):
+    """emoncms integration UI config flow."""
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: ConfigEntry,
+    ) -> OptionsFlowWithConfigEntry:
+        """Get the options flow for this handler."""
+        return EmoncmsOptionsFlow(config_entry)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Initiate a flow via the UI."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            self._async_abort_entries_match(
+                {
+                    CONF_API_KEY: user_input[CONF_API_KEY],
+                    CONF_URL: user_input[CONF_URL],
+                }
+            )
+            result = await get_feed_list(
+                self.hass, user_input[CONF_URL], user_input[CONF_API_KEY]
+            )
+            if not result[CONF_SUCCESS]:
+                errors["base"] = result[CONF_MESSAGE]
+            else:
+                title = sensor_name(user_input[CONF_URL])
+                return self.async_create_entry(title=title, data=user_input)
+        return self.async_show_form(
+            step_id="user",
+            data_schema=self.add_suggested_values_to_schema(
+                vol.Schema(
+                    {
+                        vol.Required(CONF_URL): str,
+                        vol.Required(CONF_API_KEY): str,
+                    }
+                ),
+                user_input,
+            ),
+            errors=errors,
+        )
+
+    async def async_step_import(self, import_info: ConfigType) -> ConfigFlowResult:
+        """Import config from yaml."""
+        url = import_info[CONF_URL]
+        api_key = import_info[CONF_API_KEY]
+        include_only_feeds = None
+        if import_info.get(CONF_ONLY_INCLUDE_FEEDID) is not None:
+            include_only_feeds = list(map(str, import_info[CONF_ONLY_INCLUDE_FEEDID]))
+        if not include_only_feeds:
+            emoncms_result = await get_feed_list(self.hass, url, api_key)
+            if emoncms_result[CONF_SUCCESS]:
+                include_only_feeds = [
+                    feed[FEED_ID] for feed in emoncms_result[CONF_MESSAGE]
+                ]
```

Comment:
I think I should remove these lines so that when yaml is imported, feed list is imported from the yaml only if `include_only_feed_id` is present ? These lines are used to import **all the feeds** when the user has no `include_only_feed_id` in his yaml configuration....but we dont really want this as we want the user to choose with the select so that he can import only what he really needs....

Reply:
done